src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6278,17 +6290,30 @@ private ScanTypeFlags ScanTupleType(out SyntaxToken lastTokenOfType)
                             lastTokenOfType = this.EatToken();
                             return ScanTypeFlags.NotType;
                         }
+                        else if (tupleElementType == ScanTypeFlags.MustBeType)
+                        {
+                            mustBeType = true;
```

Comment:
The more I think about this, I think I'll need to remove this code anyways for deconstruction-declaration.